### DISCUSSION TITLE
Installing Bookworm on RPI4: some odd results.

### DISCUSSION THREAD

**Discussion by feh123** [RaspAP/raspap-webgui#1477] [General]
Over the last few weeks I have been looking to update RaspAP to 3.0 using the new RPI OS Bookworm. I have two RPI4 (4GB). One has a PoE hat, the other has no PoE. The PoE device I setup to just connect via eth0 and broadcast wlan0. The other device has installed a usb wifi adaptor (Netgear A6210-100PES). Both these RPI4's are several years old and have worked really well with these configurations, updating was always straight forward as new versions of the software were published. They have been constant and essential travelling companions with me.
With the arrival of Bookworm I have hit two problems. I do not understand what causes them but it is most likely something specific to my setup and how I update them. But I thought I would share the two major solutions I have found to get the devices to run 3.0 versions.
1 For my RPI PoE eth0 setup I install the software (completely following the quick installer menu including adding the country code in raspi-config). However I always have to separately add the country code to hostapd.conf after the installation as the country code line is empty. The device then works with Bookworm and v3.0.
2 For my RPI4 wlan0/wlan1 setup for all my attempts I cannot get this to work on Bookworm. Bullseye installs v3.0 without a problem.
So this is where I am - both devices work fine so it's not an issue - I just thought I would document my rather odd results. One day I hope to be able to explain what I am doing that produces them.

**Comment by billz** [comment]
@feh123 RaspAP's debug log generator was created specifically for these cases.

**Comment by feh123** [comment]
Thanks @billz I will try this out when I update to 3.0.4.
